keysManagement/keysHolder.go

```diff
@@ -153,6 +161,89 @@ func (holder *virtualPeersHolder) GetMachineID(pkBytes []byte) (string, error) {
 	return pInfo.machineID, nil
 }
 
+// IncrementRoundsWithoutReceivedMessages increments the number of rounds without received messages on a provided public key
+func (holder *virtualPeersHolder) IncrementRoundsWithoutReceivedMessages(pkBytes []byte) error {
+	if holder.isMainMachine {
+		return nil
+	}
+
+	pInfo := holder.getPeerInfo(pkBytes)
+	if pInfo == nil {
+		return fmt.Errorf("%w in IncrementRoundsWithoutReceivedMessages for public key %s",
+			errMissingPublicKeyDefinition, hex.EncodeToString(pkBytes))
+	}
+
+	pInfo.incrementRoundsWithoutReceivedMessages()
+
+	return nil
+}
+
+// ResetRoundsWithoutReceivedMessages resets the number of rounds without received messages on a provided public key
+func (holder *virtualPeersHolder) ResetRoundsWithoutReceivedMessages(pkBytes []byte) error {
+	if holder.isMainMachine {
+		return nil
+	}
+
+	pInfo := holder.getPeerInfo(pkBytes)
+	if pInfo == nil {
+		return fmt.Errorf("%w in ResetRoundsWithoutReceivedMessages for public key %s",
+			errMissingPublicKeyDefinition, hex.EncodeToString(pkBytes))
+	}
+
+	pInfo.resetRoundsWithoutReceivedMessages()
+
+	return nil
+}
+
+// GetAllManagedKeys returns all keys that will be managed by this node
+func (holder *virtualPeersHolder) GetAllManagedKeys() map[string]crypto.PrivateKey {
+	holder.mut.RLock()
+	defer holder.mut.RUnlock()
+
+	allManagedKeys := make(map[string]crypto.PrivateKey)
+	for pk, pInfo := range holder.data {
```

Comment:
`holder.isMainMachine` can be extracted in a var outside the for loop - slight improvement

Reply:
not quite an improvement as it will be also a var. `holder.isMainMachine` is already a bool var